### ISSUE TITLE
Fetch API cannot load file:///android_asset/www/js/app.js. URL scheme "file" is not supported.

### ISSUE THREAD

**Issue by aetonsi** [xpl/get-source#6]
Hello again,
i'd like to know if it would be possible to use and xhr request instead of fetch for `fetchFileAsync`.

Fetch only supports the `http` protocol, while xhr also supports `file` for compiled apps on android & similar.
I tried to replace it with an xhr request, and the module seems to be working.

This would be the code (very similar to the synchronous function):
```js
module.exports.async = impl (function fetchFileAsync (path) {
  return new Promise ((resolve, reject) => {
    if (isBrowser) {
        let xhr = new XMLHttpRequest ()
        xhr.open ('GET', path)
        xhr.send (null)
        xhr.onload = r => resolve(xhr.responseText)
    } else {
        module.require ('fs').readFile (path, { encoding: 'utf8' }, (e, x) => {
          e ? reject (e) : resolve (x)
        })
    }
  })
})
```

Is there any reason why fetch would be necessarily needed?

<hr>

Have a nice day

**Comment by xpl** [comment]
> Is there any reason why fetch would be necessarily needed?

Not really, let's make it an XHR request. I didn't know that there is a difference in behavior.

**Comment by xpl** [comment]
Can you make it a PR? Also, what about error handling? :)

**Comment by aetonsi** [comment]
> Can you make it a PR? Also, what about error handling? :)

Done, should be good

**Comment by xpl** [comment]
Thanks! It's been merged and also available in Stacktracey since 2.0.14

**Comment by aetonsi** [comment]
Thank you, and have a nice day.
Closing
